core/trino-main/src/main/java/io/trino/server/ui/MultipartUiCookie.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.server.ui;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Splitter;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import jakarta.ws.rs.core.Cookie;
+import jakarta.ws.rs.core.NewCookie;
+
+import java.time.Instant;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static jakarta.ws.rs.core.NewCookie.DEFAULT_MAX_AGE;
+import static java.util.Objects.requireNonNull;
+
+public class MultipartUiCookie
+{
+    // https://chromestatus.com/feature/4946713618939904
+    private static final int COOKIE_LENGTH_LIMIT = 4096;
+
+    private final String cookieName;
+    private final String location;
+
+    public MultipartUiCookie(String cookieName, String location)
+    {
+        this.cookieName = requireNonNull(cookieName, "cookieName is null");
+        this.location = requireNonNull(location, "location is null");
+    }
+
+    public NewCookie[] create(String token, Instant tokenExpiration, boolean isSecure)
+    {
+        Date expiration = Optional.ofNullable(tokenExpiration).map(Date::from).orElse(null);
+        ImmutableList.Builder<NewCookie> cookiesToSet = ImmutableList.builder();
+        int index = 0;
+        for (String part : splitValueByLength(token)) {
+            cookiesToSet.add(new NewCookie.Builder(cookieName(index++))
+                    .value(part)
+                    .path(location)
+                    .domain(null)
+                    .comment(null)
+                    .maxAge(DEFAULT_MAX_AGE)
+                    .expiry(expiration)
+                    .secure(isSecure)
+                    .httpOnly(true)
+                    .build());
+        }
+        return cookiesToSet.build().toArray(new NewCookie[0]);
+    }
+
+    public Optional<String> read(Map<String, Cookie> existingCookies)
+    {
+        long cookiesCount = existingCookies.values().stream()
+                .filter(cookie -> cookie.getName().startsWith(cookieName))
```

Comment:
possible name collision? probably better to use regex

Reply:
Right, fixed